build.gradle

```diff
@@ -59,4 +65,95 @@ subprojects {
   }
 
   apply from: "${rootDir}/gradle/dependencies.gradle"
+
+  plugins.withType(MavenPublishPlugin).withType(SigningPlugin) {
+    apply plugin: "org.shipkit.travis"
+    apply plugin: "org.shipkit.github-pom-contributors"
+    apply plugin: "org.shipkit.release-notes"
+    apply plugin: "org.shipkit.compare-publications"
+    publishing {
+      publications {
+        tasks.downloadPreviousReleaseArtifacts.previousSourcesJarUrl =
+            "https://repo1.maven.org/maven2/com/linkedin/coral/" + subproject.name + "/" + tasks.fetchReleaseNotes.previousVersion +
+            subproject.name + "-" + tasks.fetchReleaseNotes.previousVersion + "-sources.jar"
+        maven(MavenPublication) { publication ->
+          tasks.performRelease.dependsOn(tasks.publish)
+          from components.java
+          artifact(javadocJar) {
+            classifier = 'javadoc'
+          }
+          artifact(sourcesJar) {
+            classifier = 'sources'
+          }
```

Comment:
If your modules are all using the `java` plugin, you should be able to use this block:
```groovy
java {
    withSourcesJar()
    withJavadocJar()
}
```
in each of your modules (or in this file, outside the `publications` block) and then source and javadoc jars will get added to the publication automatically since you're using `from components.java`